modules/build/src/test/scala/scala/build/tests/SourceGeneratorTests.scala

```diff
@@ -0,0 +1,383 @@
+             |    val scalacOptions = Seq("-Xasync")
+             |    val scalaCompilerPlugins = Seq("org.wartremover:wartremover_3.2.2:3.0.9")
+             |    val dependencies = Seq("com.lihaoyi:os-lib_3:0.9.1")
+             |    val resolvers = Seq("ivy:file:.../scala-cli-tests-extra-repo/local-repo/...", "https://repo1.maven.org/maven2", "ivy:file:.../.ivy2/local/")
+             |    val resourceDirs = Seq("${root / "resources"}")
+             |    val customJarsDecls = Seq("${root / "TEST1.jar"}", "${root / "TEST2.jar"}")
+             |  }
+             |
+             |  object TestScopeInfo {
+             |    val sources = Nil
+             |    val scalacOptions = Nil
+             |    val scalaCompilerPlugins = Nil
+             |    val dependencies = Nil
+             |    val resolvers = Nil
+             |    val resourceDirs = Nil
+             |    val customJarsDecls = Nil
+             |  }
+             |
+             |}
+             |""".stripMargin)
+    }
+  }
+
+  test(s"BuildInfo for Scala 2") {
+    val inputs = TestInputs(
+      os.rel / "main.scala" ->
+        s"""//> using dep "com.lihaoyi::os-lib:0.9.1"
+           |//> using option "-Xasync"
+           |//> using plugin "org.wartremover:::wartremover:3.0.9"
+           |//> using scala 2.13.6
+           |//> using jvm 11
+           |//> using mainClass "Main"
+           |//> using resourceDir ./resources
+           |//> using jar TEST1.jar TEST2.jar
+           |//> using computeVersion "command:echo TestVersion"
+           |
+           |//> using buildInfo
+           |
+           |import scala.cli.build.BuildInfo
+           |
+           |object Main extends App {
+           |  println(s"Scala version: $${BuildInfo.scalaVersion}")
+           |  BuildInfo.MainScopeInfo.customJarsDecls.foreach(println)
+           |}
+           |""".stripMargin
+    )
+
+    inputs.withBuild(baseOptions, buildThreads, bloopConfigOpt) {
+      (root, _, maybeBuild) =>
+        expect(maybeBuild.orThrow.success)
+        val projectDir = os.list(root / ".scala-build").filter(
+          _.baseName.startsWith(root.baseName + "_")
+        )
+        expect(projectDir.size == 1)
+        val buildInfoPath = projectDir.head / "src_generated" / "main" / "BuildInfo.scala"
+        expect(os.isFile(buildInfoPath))
+
+        val buildInfoContent = os.read(buildInfoPath)
+
+        expect(normalizeResolvers(buildInfoContent) ==
+          s"""package scala.cli.build
+             |
+             |object BuildInfo {
+             |  val scalaVersion = "2.13.6"
+             |  val platform = "JVM"
+             |  val jvmVersion = Some("11")
+             |  val scalaJsVersion = None
+             |  val jsEsVersion = None
+             |  val scalaNativeVersion = None
+             |  val mainClass = Some("Main")
+             |  val projectVersion = Some("TestVersion")
+             |
+             |
+             |  object MainScopeInfo {
```

Comment:
```suggestion
             |  object Main {
```
wouldn't `Main` and `Test` be enough? `BuildInfo.Main` already seems clear. Or at least drop the `Info` suffix

Reply:
I think that `BuildInfo.Main` should be sufficient.